### PR TITLE
SERVICES-1120 add featured collections mutations; use CacheInfo inste…

### PR DESCRIPTION
**Pull request by johnykes**
- add & remove mutations for featured collections with cache invalidation (also set featured collections `identifier` as unique field & split cache login into a separate service)
- replace hard-coded values with CacheInfo key & ttl

### CODE REVIEW

File: src/common/persistence/persistence.service.ts [hunk 1]
```diff
@@ -316,12 +317,36 @@ export class PersistenceService {
     limit: number = 20,
     offset: number = 0,
   ): Promise<[FeaturedCollectionEntity[], number]> {
```
Comment (danielailie) [inline src/common/persistence/persistence.service.ts:319]: can you remove the console.log

File: src/db/featuredNfts/featured.entity.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export class FeaturedCollectionEntity {
   @PrimaryGeneratedColumn()
   id: number;
   @Column({ length: 25 })
```
Comment (danielailie) [inline src/db/featuredNfts/featured.entity.ts:21]: here we should not have unique, we can have a collection of type featured and also hero
the unique condition should be on both columns (identifier and type)

File: src/modules/featured/featured.service.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ export class FeaturedService {
         this.redisClient,
         cacheKey,
         getAssetLiked,
```
Comment (danielailie) [inline src/modules/featured/featured.service.ts:38]: now that we invalidate the cache we could use a higher ttl like oneDay or mode

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/modules/featured/featured.service.ts (hunk 3 above)
Comment (johnykes) [reply src/modules/featured/featured.service.ts:38]: Done

File: src/db/featuredNfts/featured.entity.ts (hunk 2 above)
Comment (johnykes) [reply src/db/featuredNfts/featured.entity.ts]: Done

File: src/common/persistence/persistence.service.ts (hunk 1 above)
Comment (johnykes) [reply src/common/persistence/persistence.service.ts]: Done

File: src/modules/featured/featured.service.ts [hunk 4]
```diff
@@ -94,7 +86,38 @@ export class FeaturedService {
...
+      collection,
+    );
+    if (isRemoved) {
+      await this.invalidateFeaturedCollectionsCache();
+    }
+    return isRemoved;
+  }
+
+  async invalidateFeaturedCollectionsCache(): Promise<void> {
+    await Promise.all([
```
Comment (danielailie) [inline src/modules/featured/featured.service.ts:114]: this in not needed here, and maybe rename this method trigger cache invalidation and keep only the trigger event.

**Review by danielailie — DISMISSED** [review]

File: src/modules/featured/featured.service.ts [hunk 5]
```diff
@@ -94,7 +98,42 @@ export class FeaturedService {
...
-  private getFeaturedCollectionsCacheKey() {
-    return generateCacheKeyFromParams('featuredCollections');
+  async addFeaturedCollection(
+    collection: string,
+    type: FeaturedCollectionTypeEnum,
+  ): Promise<boolean> {
+    const isAdded = await this.persistenceService.addFeaturedCollection(
+      collection,
+      type,
+    );
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/featured/featured.service.ts:96]: Done

File: src/modules/featured/featured.service.ts (hunk 4 above)
Comment (johnykes) [reply src/modules/featured/featured.service.ts]: Done

**Review by gabrielmatei — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Review by gabrielmatei — APPROVED** [review]

**Pull request merged**